### PR TITLE
Fix cmake 3.12 support

### PR DESCRIPTION
**Pull request by benson31**
NVCC throws a tantrum if you pass the `libcublas_device.a` library twice on the device-link line. This avoids that.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -209,6 +181,20 @@ if (NOT OpenCV_FOUND)
 endif ()
 set(LBANN_HAS_OPENCV ${OpenCV_FOUND})
 
+# CUDA-ness of LBANN is 1:1 with Hydrogen. Iff Hydrogen has CUDA, LBANN gets CUDA.
```
Comment (bvanessen) [inline CMakeLists.txt]: Do you really mean if-and-only-if hydrogen has CUDA, or is the iff just a typo.

File: CMakeLists.txt [hunk 2]
```diff
@@ -416,21 +401,6 @@ if (LBANN_HAS_ALUMINUM)
...
-  endif ()
-  target_link_libraries(lbann PUBLIC ${cuBLAS_LIBRARIES})
-
-  if(LBANN_HAS_NCCL2)
-    target_link_libraries(lbann PUBLIC cuda::nccl)
-  endif()
-endif()
-if(LBANN_HAS_CUDNN)
-  target_link_libraries(lbann PUBLIC ${cuDNN_LIBRARIES})
-endif ()
```
Comment (bvanessen) [inline CMakeLists.txt]: Are these getting included elsewhere and that is why they are removed from here?

File: CMakeLists.txt (hunk 1 above)
Comment (benson31) [reply CMakeLists.txt]: I mean if and only if. #mathematicianed.

File: CMakeLists.txt (hunk 2 above)
Comment (benson31) [reply CMakeLists.txt]: They all get added to the cuda::toolkit IMPORTED target.

**Comment by bvanessen** [comment]
Looks good, but I haven't tested it yet.

**Comment by benson31** [comment]
Builds on pascal with cuda 9.2 and cuda 10.0 with appropriate Hydrogen branch. Builds on Lassen with gcc/7.3.1 and cuda 9.2.

**Comment by benson31** [comment]
IMO, this is good to go now.

**Comment by benson31** [comment]
Just merged LLNL/elemental#64 so this should build with CUDA 9 or CUDA 10 out-of-the-box.

**Pull request merged**